src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -41,15 +41,67 @@ public static partial class AIFunctionFactory
+    ///     </description>
+    ///   </item>
+    ///   <item>
+    ///     <description>
+    ///       By default, <see cref="IServiceProvider"/> parameters are bound from the <see cref="AIFunctionArguments.Services"/> property
+    ///       and are not included in the JSON schema. If the parameter is optional, such that a default value is provided,
+    ///       <see cref="AIFunctionArguments.Services"/> is allowed to be <see langword="null"/>; otherwise, <see cref="AIFunctionArguments.Services"/>
+    ///       must be non-<see langword="null"/>, or else the invocation will fail with an exception due to the required nature of the parameter.
+    ///       The handling of <see cref="IServiceProvider"/> parameters may be overridden via <see cref="AIFunctionFactoryOptions.ConfigureParameterBinding"/>.
+    ///     </description>
+    ///   </item>
+    ///   <item>
+    ///     <description>
+    ///       By default, <see cref="AIFunctionArguments"/> parameters are bound directly to <see cref="AIFunctionArguments"/> instance
+    ///       passed into <see cref="AIFunction.InvokeAsync"/> and are not included in the JSON schema. If the <see cref="AIFunctionArguments"/>
+    ///       instance passed to <see cref="AIFunction.InvokeAsync"/> is <see langword="null"/>, the <see cref="AIFunction"/> implementation
+    ///       manufactures an empty instance, such that parameters of type <see cref="AIFunctionArguments"/> may always be satisfied, whether
+    ///       optional or not. The handling of <see cref="AIFunctionArguments"/> parameters may be overridden via
+    ///       <see cref="AIFunctionFactoryOptions.ConfigureParameterBinding"/>.
+    ///     </description>
+    ///   </item>
+    /// </list>
+    /// All other parameter types are, by default, bound from the <see cref="AIFunctionArguments"/> dictionary passed into <see cref="AIFunction.InvokeAsync"/>
+    /// and are included in the generated JSON schema. This may be overridden by the <see cref="AIFunctionFactoryOptions.ConfigureParameterBinding"/> provided
+    /// via the <paramref name="options"/> parameter; for every parameter, the delegate is enabled to choose if the parameter should be included in the
+    /// generated schema and how its value should be bound, including handling of optionality (by default, required parameters that are not included in the
+    /// <see cref="AIFunctionArguments"/> dictionary will result in an exception being thrown). Loosely-typed additional context information may be passed
+    /// into <see cref="AIFunction.InvokeAsync"/> via the <see cref="AIFunctionArguments"/>'s <see cref="AIFunctionArguments.Context"/> dictionary; the default
+    /// binding ignores this collection, but a custom binding supplied via <see cref="AIFunctionFactoryOptions.ConfigureParameterBinding"/> may choose to
+    /// source arguments from this data.
```

Comment:
All the info here looks good and correct, and I understand having it is an outcome from the TM review.

I wonder a bit if XML docs are the right place for this level of detail though. It's not common for us to describe so much of the workings of the whole system inside XML docs is it? I was expecting https://learn.microsoft.com/en-us/dotnet/ai/ai-extensions to cover the end-to-end system while leaving XML docs on specific .NET types to be more tightly scoped to those types alone.

Reply:
> I wonder a bit if XML docs are the right place for this level of detail though

I think it is. Most of this is about what's permitted / how inputs/outputs are handled. That's a key thing devs need to understand when using AIFunctionFactory, and one of the main thing API docs cover in general (method signature-related stuff).

Reply:
OK, fine with me